docs/builtin-tools.md

```diff
@@ -419,6 +420,147 @@ print(result.output)
+            url='https://api.githubcopilot.com/mcp/',
+            authorization_token=os.getenv('GITHUB_ACCESS_TOKEN', 'mock-access-token'),
+            allowed_tools=['search_repositories', 'list_commits'],
+        )
+    ]
+)
+
+result = agent.run_sync('Give me some examples of my products.')
+print(result.output)
+#> Here are some examples of my data: Pen, Paper, Pencil.
+```
+
+_(This example is complete, it can be run "as is")_
+
+With OpenAI, you must use their responses API to access the MCP server tool.
+
+```py {title="mcp_server_openai.py"}
+import os
+
+from pydantic_ai import Agent, MCPServerTool
+
+agent = Agent(
+    'openai-responses:gpt-4o',
+    builtin_tools=[
+        MCPServerTool(
+            id='your-mcp-server',
+            url='https://api.githubcopilot.com/mcp/',
+            authorization_token=os.getenv('GITHUB_ACCESS_TOKEN', 'mock-access-token'),
+            allowed_tools=['search_repositories', 'list_commits'],
+        )
+    ]
+)
+
+result = agent.run_sync('Give me some examples of my products.')
+print(result.output)
+#> Here are some examples of my data: Pen, Paper, Pencil.
+```
+
+_(This example is complete, it can be run "as is")_
+
+### Configuration Options
+
+The `MCPServerTool` supports several configuration parameters for custom MCP servers:
+
+```py {title="mcp_server_configured_url.py"}
+import os
+
+from pydantic_ai import Agent, MCPServerTool
+
+agent = Agent(
+    'openai-responses:gpt-4o',
+    builtin_tools=[
+        MCPServerTool(
+            id='your-mcp-server',  # required field
+            url='https://api.githubcopilot.com/mcp/',  # optional field, use `url` or `provider_metadata`
+            authorization_token=os.getenv('GITHUB_ACCESS_TOKEN', 'mock-access-token'),  # optional field
+            allowed_tools=['search_repositories', 'list_commits'],  # optional field
+            description='Your MCP Server',  # optional field
+            headers={'X-CUSTOM-HEADER': 'custom-value'},  # optional field
+        )
+    ]
+)
+
+result = agent.run_sync('Give me some examples of my products.')
+print(result.output)
+#> Here are some examples of my data: Pen, Paper, Pencil.
+```
+
+For OpenAI Responses, you can use connector ID instead of URL:
+
+_(This example is complete, it can be run "as is")_
+
+```py {title="mcp_server_configured_connector_id.py"}
+import os
+
+from pydantic_ai import Agent, MCPServerTool
+
+agent = Agent(
+    'openai-responses:gpt-4o',
+    builtin_tools=[
+        MCPServerTool(
+            id='your-mcp-server',  # required field
+            url='x-openai-connector:connector_googlecalendar',  # required field
+            authorization_token=os.getenv('GITHUB_ACCESS_TOKEN', 'mock-access-token'),  # optional field
+            allowed_tools=['search_repositories', 'list_commits'],  # optional field
+            description='Your MCP Server',  # optional field
+            headers={'X-CUSTOM-HEADER': 'custom-value'},  # optional field
```

Comment:
We can remove this required/optional comments if we modify the first example to only use the required ones.